packages/journey-manager/src/index.ts

```diff
@@ -126,6 +136,29 @@ const matchesUrlCondition = (urlCondition: UrlCondition): boolean => {
   return false;
 };
 
+const withElements = async (
+  steps: Array<StepWithQuery>,
+): Promise<StepWithQueryAndElements[]> => {
+  const targets = await Promise.all(
+    steps
+      .filter(
+        ({ urlCondition }) =>
+          urlCondition == null || matchesUrlCondition(urlCondition),
+      )
+      .map(async (step) => {
+        try {
+          return {
+            ...step,
+            elements: await find(step.query),
```

Comment:
For appear on mutation observer, I'd imagine you want the sync version rather than waiting up to a second for the element to appear.